third_party/proton/test/test_api.py

```diff
@@ -28,146 +29,147 @@ def test_profile():
     pathlib.Path("test.hatchet").unlink()
 
 
-def test_profile_decorator():
-    f = tempfile.NamedTemporaryFile(delete=True)
-    name = f.name.split(".")[0]
+def test_profile_decorator(tmp_path):
+    temp_file = tmp_path / "test_profile_decorator.hatchet"
 
-    @proton.profile(name=name)
+    @proton.profile(name=str(temp_file).split(".")[0])
```

Comment:
Pre-existing, but this should be a more robust way to remove the extension. Here and everwhere else this pattern is used.
```suggestion
    @proton.profile(name=str(temp_file.with_suffix("")))
```